pkg/kubelet/cpumanager/cpu_manager.go

```diff
@@ -273,3 +304,61 @@ func (m *manager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitR
 		Message: "Node is experiencing CPU Pressure",
 	}
 }
+
+const (
+	reason  = "Evicted"
+	message = "The node no longer has CPUs available to run the pod"
+)
+
+func isNoCPUPod(pod *v1.Pod) bool {
+	for _, container := range pod.Spec.Containers {
+		if isNoCPUContainer(&container) {
+			return true
+		}
+	}
+	return false
+}
+
+func isNoCPUContainer(container *v1.Container) bool {
+	for resourceName, request := range container.Resources.Requests {
+		if resourceName == v1.ResourceCPU && !request.IsZero() {
+			return false
+		}
+	}
+	return true
+}
+
+func getNoCPUPods(pods []*v1.Pod) []*v1.Pod {
+	var noCPUPods []*v1.Pod
+	for _, pod := range pods {
+		if isNoCPUPod(pod) {
+			noCPUPods = append(noCPUPods, pod)
+		}
+	}
+	return noCPUPods
+}
+
+func (m *manager) evictNoCPUPods() error {
+	noCPUPods := getNoCPUPods(m.getPodsFunc())
+	if len(noCPUPods) == 0 {
+		return nil
+	}
+
+	glog.Infof("[cpumanager] attempting to evict pods with no CPU request because shared CPU pool is empty: %v", noCPUPods)
+	for _, pod := range noCPUPods {
+		status := v1.PodStatus{
+			Phase:   v1.PodFailed,
+			Message: message,
+			Reason:  reason,
+		}
+		// record that we are evicting the pod
+		m.recorder.Eventf(pod, v1.EventTypeWarning, reason, message)
+		// this is a blocking call and should only return when the pod and its containers are killed.
+		err := m.killPodFunc(pod, status, nil)
+		if err != nil {
+			return fmt.Errorf("[cpumanager]: pod %s failed to evict %v", format.Pod(pod), err)
```

Comment:
Could we log and continue instead? Otherwise one container that the CRI can't evict for some reason could block the rest of the evictions.